pkg/kn/config/config.go

```diff
@@ -119,17 +126,41 @@ func BootstrapConfig() error {
 		return err
 	}
 
+	if globalConfig.skipInitCfg && globalConfig.reInitCfg {
+		return fmt.Errorf("only one of --no-init and --init may be specified")
+	}
+
+	if globalConfig.skipInitCfg {
+		return nil
+	}
+
 	// Check if configfile exists. If not, just return
 	configFile := GlobalConfig.ConfigFile()
 	_, err = os.Lstat(configFile)
-	if err != nil {
-		if os.IsNotExist(err) {
-			// No config file to read
-			return nil
+
+	if err != nil && os.IsNotExist(err) {
+		cfgDir := filepath.Dir(configFile)
+		if err := os.MkdirAll(cfgDir, 0775); err != nil && !os.IsExist(err) {
+			return fmt.Errorf("Failed creating directory path %s: %w", cfgDir, err)
+		}
+		if err := viper.SafeWriteConfigAs(configFile); err != nil {
+			return fmt.Errorf("Failed saving initial file %s: %w", configFile, err)
 		}
+		return nil
+	} else if err != nil {
 		return fmt.Errorf("cannot stat configfile %s: %w", configFile, err)
 	}
 
+	if globalConfig.reInitCfg {
+		bkupConfigFile := fmt.Sprintf("%s.bkup.%s", configFile, time.Now().Format(time.RFC3339))
+		if err := os.Rename(configFile, bkupConfigFile); err != nil {
+			return fmt.Errorf("Failed storing configuration file (%s) as backup: %w", configFile, err)
+		}
+		if err := viper.SafeWriteConfigAs(configFile); err != nil {
```

Comment:
How are the default values determined for the new config file?

For example when I tried to overwrite my local config file, it ends up with empty plugin dir and `path-lookup: false` that's opposite what we're now using as default value.

```console
➜  client git:(closes_1232) kn plugin list --init
No plugins found.
➜  client git:(closes_1232) bat ~/.config/kn/config.yaml
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /home/dsimansk/.config/kn/config.yaml
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ plugins:
   2   │   directory: ""
   3   │   path-lookup: false
───────┴──────────────────────────
```

Reply:
In addition is there're a cobra/viper way to create config file from a template. IMO it'd useful to have a few comments explaining the parameters for the users.